app/scripts/migrations/054.js

```diff
@@ -0,0 +1,57 @@
+import { cloneDeep } from 'lodash';
+
+const version = 54;
+
+/**
+ * Remove tokens that don't have an address due to
+ * lack of previous addToken validation.  Also removes
+ * an unwanted, undefined image property
+ */
+export default {
+  version,
+  async migrate(originalVersionedData) {
+    const versionedData = cloneDeep(originalVersionedData);
+    versionedData.meta.version = version;
+
+    const { PreferencesController } = versionedData.data;
+
+    // These properties are arrays and can be directly filtered
+    const tokenArrays = ['tokens', 'hiddenTokens'];
+    tokenArrays.forEach((tokenKey) => {
+      if (PreferencesController[tokenKey]) {
```

Comment:
This `if` is required because neither `PreferencesController.tokens` or `PreferencesController. hiddenTokens` exists in the migration data to this point.

Reply:
What do you mean by "neither PreferencesController.tokens or PreferencesController. hiddenTokens exists in the migration data to this point"? If they don't exist, is this `forEach` necessary? Will the code inside the if block ever run?

Perhaps you meant that it is possible for PreferencesController.tokens or PreferencesController.hiddenTokens to not exist at this point.